src/main/java/com/google/sps/servlets/RoutesStoring.java

```diff
@@ -118,19 +119,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       datastore.put(routeEntity);
 
       if (routeStatus != RouteStatus.COPY) {
-        for (long userId : editorsArray) {
-          Entity linkEntity = new Entity("RouteUserLink", KeyFactory.createKey("User", userId));
-          linkEntity.setProperty("routeId", routeEntity.getKey().getId());
-          linkEntity.setProperty("userAccess", UserAccessType.EDITOR);
-          // Add entity for editor.
-          datastore.put(linkEntity);
+        for (String friendCode : editorsArray) {
+          Filter friendCodeFilter =
+              new FilterPredicate("friendCode", FilterOperator.EQUAL, friendCode);
+          Query friendQuery = new Query("User").setFilter(friendCodeFilter);
+          List<Entity> friend =
+              datastore.prepare(friendQuery).asList(FetchOptions.Builder.withDefaults());
+
+          if (friend.size() == 1) {
```

Comment:
Why is this a special case?
Please add a comment

Reply:
Done